### PR TITLE
[samples] Samples bugfixes and new features

### PR DESCRIPTION
**Pull request by fzhar**

### CODE REVIEW

File: samples/sample_vpp/src/sample_vpp.cpp [hunk 1]
```diff
@@ -301,22 +300,13 @@ int main(int argc, msdk_char *argv[])
 
     SurfaceVPPStore     surfStore;
 
-    unique_ptr<PTSMaker> ptsMaker;
```
Comment (onabiull) [inline samples/sample_vpp/src/sample_vpp.cpp]: Please do not revert previous work.

File: samples/sample_vpp/src/sample_vpp.cpp [hunk 2]
```diff
@@ -301,22 +300,13 @@ int main(int argc, msdk_char *argv[])
...
+    auto_ptr<PTSMaker> ptsMaker;
 
     /* generators for ROI testing */
     ROIGenerator       inROIGenerator;
     ROIGenerator       outROIGenerator;
     bool               bROITest[2] = {false, false};
 
-#ifdef ENABLE_VPP_RUNTIME_HSBC
```
Comment (onabiull) [inline samples/sample_vpp/src/sample_vpp.cpp]: HSBC should remain here

File: samples/sample_vpp/src/sample_vpp.cpp [hunk 3]
```diff
@@ -783,78 +777,6 @@ int main(int argc, msdk_char *argv[])
 
                 }
 #endif
-
```
Comment (onabiull) [inline samples/sample_vpp/src/sample_vpp.cpp]: ... and here

File: samples/sample_vpp/src/sample_vpp.cpp [hunk 4]
```diff
@@ -971,77 +893,6 @@ int main(int argc, msdk_char *argv[])
             }
             else
             {
-#ifdef ENABLE_VPP_RUNTIME_HSBC
```
Comment (onabiull) [inline samples/sample_vpp/src/sample_vpp.cpp]: ... and here

File: samples/sample_vpp/src/sample_vpp_parser.cpp [hunk 5]
```diff
@@ -64,11 +64,7 @@ msdk_printf(MSDK_STRING("   [-scrY  y]                  - cropY  of src video (d
 msdk_printf(MSDK_STRING("   [-scrW  w]                  - cropW  of src video (def: width)\n"));
 msdk_printf(MSDK_STRING("   [-scrH  h]                  - cropH  of src video (def: height)\n"));
 msdk_printf(MSDK_STRING("   [-sf   frameRate]           - frame rate of src video (def: 30.0)\n"));
-#if MFX_VERSION >= MFX_VERSION_NEXT
```
Comment (onabiull) [inline samples/sample_vpp/src/sample_vpp_parser.cpp]: Please don't remove VERSION_NEXT-guarded parts

File: samples/sample_vpp/src/sample_vpp_parser.cpp [hunk 6]
```diff
@@ -199,28 +198,6 @@ msdk_printf(MSDK_STRING("   [-pa_hue  hue]        - procamp hue property.
 msdk_printf(MSDK_STRING("   [-pa_sat  saturation] - procamp satursation property. range [   0.0,  10.0] (def: 1.0)\n"));
 msdk_printf(MSDK_STRING("   [-pa_con  contrast]   - procamp contrast property.    range [   0.0,  10.0] (def: 1.0)\n"));
 msdk_printf(MSDK_STRING("   [-pa_bri  brightness] - procamp brightness property.  range [-100.0, 100.0] (def: 0.0)\n\n"));
-#ifdef ENABLE_VPP_RUNTIME_HSBC
```
Comment (onabiull) [inline samples/sample_vpp/src/sample_vpp_parser.cpp]: ... and here.

File: samples/sample_vpp/src/sample_vpp_parser.cpp [hunk 7]
```diff
@@ -323,12 +300,6 @@ mfxU32 Str2FourCC( msdk_char* strInput )
     {
         fourcc = MFX_FOURCC_YV12;
     }
-#if (MFX_VERSION >= MFX_VERSION_NEXT)
```
Comment (onabiull) [inline samples/sample_vpp/src/sample_vpp_parser.cpp]: Please keep modified code.

File: samples/sample_vpp/src/sample_vpp_parser.cpp [hunk 8]
```diff
@@ -828,60 +799,7 @@ mfxStatus vppParseResetPar(msdk_char* strInput[], mfxU8 nArgNum, mfxU8& curArg,
                 i++;
                 msdk_sscanf(strInput[i], MSDK_STRING("%lf"), &pParams->procampParam[paramID].saturation);
             }
-#ifdef ENABLE_VPP_RUNTIME_HSBC
```
Comment (onabiull) [inline samples/sample_vpp/src/sample_vpp_parser.cpp]: ... and here.

File: samples/sample_vpp/src/sample_vpp_parser.cpp [hunk 9]
```diff
@@ -1340,60 +1284,6 @@ mfxStatus vppParseInputString(msdk_char* strInput[], mfxU8 nArgNum, sInputParams
                 i++;
                 msdk_sscanf(strInput[i], MSDK_STRING("%lf"), &pParams->procampParam[0].saturation);
             }
-#ifdef ENABLE_VPP_RUNTIME_HSBC
```
Comment (onabiull) [inline samples/sample_vpp/src/sample_vpp_parser.cpp]: ... and here

File: samples/sample_vpp/src/sample_vpp.cpp (hunk 1 above)
Comment (fzhar) [reply samples/sample_vpp/src/sample_vpp.cpp]: Not me - that's git's auto merge

**Pull request closed without merging**